src/components/colorbar/draw.js

```diff
@@ -90,7 +91,7 @@ function makeColorBarData(gd) {
 
                 if(cont && cont.showscale) {
                     var opts = cont.colorbar;
-                    opts._id = 'cb' + trace.uid;
+                    opts._id = 'cb' + trace.uid + (allowsMultiplotCbs && contName ? '-' + contName : '');
```

Comment:
This commits essentially resolves https://github.com/plotly/plotly.js/issues/3555 

I didn't add support for `marker.line.` colorbars (yet) - let me know you think that's a good idea.